### DISCUSSION TITLE
Embedded scripting for line parsing

### DISCUSSION THREAD

**Discussion by jpnorair** [Serial-Studio/Serial-Studio#39] [Ideas]
Just a note: I've forked this repo, and this week I plan to implement an option to have the input lines parsed by an external script (which is loaded into serial-studio).

Qt has built-in support for Javascript, and the core format is JSON, so Javascript makes a lot of sense for this.

My rationale is simply that it's not always possible to alter the log format of the target HW to be compliant with the Serial Studio protocol.

Thanks for the great work!  I'll send the pull request in (hopefully) about a week.

**Comment by alex-spataru** [comment]
Thanks a lot for the interest in the project! Please let me know if I can help you with something. I was kind of busy working on other issues and did not notice this comment, sorry for the late response.

**Reply by jpnorair** [reply]
No problem.  I have some working code in my fork repo.  I'm adding more today that can load data from a file or a named-pipe.  I'm not a Qt dev, so it may be ugly -- I'm just trying to build-on your existing code without breaking things.

**Comment by jpnorair** [comment]
I have the JS external script mode working (mostly), and also I added a file import method.  It should also work for named pipes.  I also improved the performance of the read loop in Manager by reducing the amount of frame copying.

The changes are visible here (yes, I need to pull the latest from upstream): 
https://github.com/jpnorair/Serial-Studio

One thing you could help with -- I am looking for an example of what auto-mode JSON input should look like.  I can't seem to get Serial Studio to plot graphs.

**Comment by alex-spataru** [comment]
Hi @jpnorair. Sorry for the extremely late response. I was fooling around the discussions section and found your comment. I think that I have changed a lot of files between your fork and the current branch, however, I would be happy to merge your changes. Please let me know if I can help you with that.

Regarding your question, you can find an example Arduino program that sends JSON in "auto" mode [here](https://github.com/Serial-Studio/Serial-Studio/issues/63#issuecomment-926043574). Again, sorry for the late response 😞
